public/app/features/browse-dashboards/components/BrowseView.tsx

```diff
@@ -81,7 +81,13 @@ export function BrowseView({ folderUID, width, height, permissions, isReadOnlyRe
 
   const handleItemSelectionChange = useCallback(
     (item: DashboardViewItem, isSelected: boolean) => {
-      dispatch(setItemSelectionState({ item, isSelected }));
+      // If selecting a folder, use the async thunk to collect all dashboards recursively
+      // When deselecting, the normal reducer will handle deselecting all children
+      if (item.kind === 'folder') {
+        dispatch(selectFolderWithAllDashboards({ folderUID: item.uid, isSelected }));
+      } else {
+        dispatch(setItemSelectionState({ item, isSelected }));
+      }
```

Comment:
Why do we need this change? This add complexity and is only necessary fo the export functionality, which won't be used as much as other actions. Plus we are already filtering out folders when doing bulk export?